### DISCUSSION TITLE
prost vs rust-protobuf

### DISCUSSION THREAD

**Discussion by dk333333** [bazelbuild/rules_rust#2582] [General]
> There are two rule sets. The first ruleset defines the rust_prost_library which generates Rust code using the [prost](https://crates.io/crates/prost) and [tonic](https://crates.io/crates/tonic) dependencies. The second ruleset defines the rust_proto_library and rust_grpc_library rules which generate Rust code using the [rust-protobuf](https://github.com/stepancheg/rust-protobuf/) dependencies.

If I understand correctly these are two distinct approaches - is one generally preferred over the other? I believe the examples only use rust-protobuf? https://github.com/bazelbuild/rules_rust/tree/main/examples/proto

**Comment by UebelAndre** [comment]
I don’t think any maintainers are using the `rust-protobuf` variant and there are issues open describing difficulties updating it. The Prost implementation was written in response to these issues and generally due to the popularity of it. If that helps :)

**Reply by dk333333** [reply]
gotcha, thanks!
